scripts/release/asset_publish.py

```diff
@@ -305,21 +294,70 @@ def validate_update_component(
             return False
 
     env = None
-    # Check if component's env exists
-    final_version = env_version + "-" + version_suffix if version_suffix else env_version
-    for version in [env_version, final_version]:
+    # Get environment
+    versions_to_try = [env_version]
+    if version_suffix:
+        versions_to_try.append(f"{env_version}-{version_suffix}")
+    for version in versions_to_try:
         if (env := get_asset_details(
             assets.AssetType.ENVIRONMENT.value, env_name, version, registry_name
         )) is not None:
-            break
+            return env['id']
 
-    if not env:
-        logger.log_error(f"Could not find the env for {component_name}")
+    logger.log_error(f"Environment {env_name} not found in {registry_name}; tried version(s) {versions_to_try}")
+    return None
+
+
+def validate_update_component(
+    spec_path: Path,
+    version_suffix: str,
+    registry_name: str,
+) -> bool:
+    """Validate and update component spec.
+
+    :param spec_path: Path of loaded component
+    :type spec_path: Path
+    :param version_suffix: version suffix
+    :type version_suffix: str
+    :param registry_name: name of the registry to create component in
+    :type registry_name: str
+    :return: True for successful validation and update
+    :rtype: bool
+    """
+    with open(spec_path) as f:
+        try:
+            component_dict = YAML().load(f)
+        except Exception:
+            logger.log_error(f"Error in loading component spec at {spec_path}")
+            return False
+
+    component_name = component_dict['name']
+    logger.print(f"Preparing component {component_name}")
+
+    # Handle command and parallel components
+    if 'environment' in component_dict:
+        # Command component
+        obj_with_env = component_dict
+    elif 'task' in component_dict and 'environment' in component_dict['task']:
+        # Parallel component
+        obj_with_env = component_dict['task']
+    else:
+        logger.log_error(f"Environment reference not found in {component_name}")
```

Comment:
I think we have components with no environment, that will fail on line # 349?

Reply:
Hmm. The current code would fail on those too. Lemme dig a bit.

Reply:
References:
- [Command component](https://learn.microsoft.com/en-us/azure/machine-learning/reference-yaml-component-command?view=azureml-api-2#yaml-syntax)
- [Parallel component](https://learn.microsoft.com/en-us/azure/machine-learning/reference-yaml-job-parallel?view=azureml-api-2#attributes-of-the-task-key)

Environment is required for both, although it can be either an asset ID or an inline specification. I don't see any inline environments specified in our repos.